Python/ceval.c

```diff
@@ -76,6 +76,7 @@ static PyObject * special_lookup(PyObject *, _Py_Identifier *);
 static int check_args_iterable(PyObject *func, PyObject *vararg);
 static void format_kwargs_error(PyObject *func, PyObject *kwargs);
 static void format_awaitable_error(PyTypeObject *, int);
+static inline void exit_thread(PyThreadState *);
```

Comment:
This helper function name needs a qualifier to indicate that it only conditionally exits the thread when the interpreter is shutting down. I'd suggest `exit_thread_if_finalizing`.